### PR TITLE
Stable diffusion pretraining improvements

### PR DESCRIPTION
**Pull request by marwan116**
Summary of changes in ascending order of commit history (first commit up top):

- [pep8 file renaming](https://github.com/anyscale/templates/pull/215/commits/3a68761543fc1e60050edbce76afda6450d28b49)
- [update filenaming references inside notebooks and readme](https://github.com/anyscale/templates/pull/215/commits/ca4b60591ec1116abf8694a857561f5570821015)
- [use a README notebook instead of markdown for better rendering](https://github.com/anyscale/templates/pull/215/commits/0c187f5139bb3bf40e96484767e37dd767a6aa24)
- [remove links to notebooks inside the table to avoid broken pages](https://github.com/anyscale/templates/pull/215/commits/0904d62c194a8535def9184f6e4062c20a939801)
- [add a reference to our stable diffusion blog post in the README](https://github.com/anyscale/templates/pull/215/commits/8bc13ddb4ed0984f6281d7854758f19beace908a)
- [make a note in README that the template runs on a small subset of the…](https://github.com/anyscale/templates/pull/215/commits/c9f992b2eee35928061952d2820776be9b726c5b) 
- [add a link back to the preprocessing guide inside preprocessing files](https://github.com/anyscale/templates/pull/215/commits/2e34984d6f18a97d38ea650b4fc1eb5b8388d719)
- [merge the Scale to 2 billion and Train with your custom data sections](https://github.com/anyscale/templates/pull/215/commits/94ade870f9095a9cce914a96d8e578301741382c)
- [use improved filenames (more concise and imply ordering)](https://github.com/anyscale/templates/pull/215/commits/075dcdacd55c29a9fcc82d9bc0e99186af7524b6)
- [generate a README.md file for compatibility with template requirements](https://github.com/anyscale/templates/pull/215/commits/7ed81c7f82edd46e0cb29ef21b31b9e1a45416e8)
- [keep code ordering and comments consistent across python scripts](https://github.com/anyscale/templates/pull/215/commits/97a673be8a280830f2cbcddacaa92bb190320c59)

Note given code changes are involved, I confirmed the template runs end-to-end successfully by cloning this branch onto [this workspace](https://console.anyscale.com/v2/cld_kvedZWag2qA8i5BjxUevf5i7/workspaces/expwrk_vlja3rlxszlsa6nxekjy2ydxm6/ses_f5tz8tznv84cbbm7591el11tvx) and executing the notebooks

### CODE REVIEW

File: templates/stable-diffusion-pretraining/01_preprocess.ipynb [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "   1. Compressing images into a latent space using a VAE encoder.\n",
     "   2. Generating text embeddings using a CLIP model.\n",
     "\n",
-    "This notebook executes a fully self-contained module, `Preprocessing.py`, that processes a small subset of the full 2 billion dataset to demonstrate the workload. You can parameterize the same module code to process the full dataset. The \"Scale to 2 billion images\" section below for a summarizes the necessary changes for read the full dataset.\n",
+    "This notebook executes a fully self-contained module, `preprocess.py`, that processes a small subset of the full 2 billion dataset to demonstrate the workload. You can parameterize the same module code to process the full dataset. The \"Scale to 2 billion images\" section below for a summarizes the necessary changes for read the full dataset.\n",
```
Comment (woshiyyya) [inline templates/stable-diffusion-pretraining/01_preprocess.ipynb:21]:
```suggestion
    "This notebook executes a fully self-contained module, `preprocess.py`, that processes a small subset of the full 2 billion dataset to demonstrate the workload. You can parameterize the same module code to process the full dataset. The \"Running production-scale data preprocessing\" section below for a summarizes the necessary changes for read the full dataset.\n",
```

File: templates/stable-diffusion-pretraining/02_train.ipynb [hunk 2]
```diff
@@ -16,7 +16,7 @@
     "2. **Build a Stable Diffusion model**: Use a network architecture called a U-Net to build the model.\n",
     "3. **Run the scalable training procedure**: Scale the base model from the previous step with Ray Train to enable running on a GPU compute cluster. \n",
     "\t  \n",
-    "This notebook executes a fully self-contained module, `Training.py`, that processes a small subset of the full 2 billion dataset on a small U-Net model to demonstrate the workload. You can parameterize the same module code to train over the full dataset. The `Scale to 2 billion images` section below summarizes the necessary changes for training over the full dataset. \n",
+    "This notebook executes a fully self-contained module, `train.py`, that processes a small subset of the full 2 billion dataset on a small U-Net model to demonstrate the workload. You can parameterize the same module code to train over the full dataset. The `Scale to 2 billion images` section below summarizes the necessary changes for training over the full dataset. \n",
```
Comment (woshiyyya) [inline templates/stable-diffusion-pretraining/02_train.ipynb:19]:
```suggestion
    "This notebook executes a fully self-contained module, `train.py`, that processes a small subset of the full 2 billion dataset on a small U-Net model to demonstrate the workload. You can parameterize the same module code to train over the full dataset. The `Running production-scale model training` section below summarizes the necessary changes for training over the full dataset. \n",
```

File: templates/stable-diffusion-pretraining/02_train.ipynb [hunk 3]
```diff
@@ -55,22 +61,7 @@
...
-     "languageId": "plaintext"
-    }
-   },
-   "source": [
-    "## Want to pre-train with custom data? 📈\n",
-    "\n",
-    "If you're looking to scale your Stable Diffusion pre-training with custom data, we're here to help 🙌 !\n",
-    "\n",
-    "👉 **[Check out this link](https://forms.gle/9aDkqAqobBctxxMa8) so we can assist you**."
+    "In terms of infrastructure, you would provision 4 instances of p4de.24xlarge instances for the entire process or use Anyscale's autoscaling capabilities to scale up and down as needed."
```
Comment (woshiyyya) [inline templates/stable-diffusion-pretraining/02_train.ipynb:64]:
```suggestion
    "In terms of infrastructure, you would provision 4 `p4de.24xlarge` instances for the entire process or use Anyscale's autoscaling capabilities to scale up and down as needed."
```

File: templates/stable-diffusion-pretraining/03_end_to_end.ipynb [hunk 4]
```diff
@@ -21,7 +21,7 @@
     "   * Training a U-Net model on the image latents and text embeddings.\n",
     "   * Generating model checkpoints and saving them to a remote storage system.\n",
     "\n",
-    "This notebook executes a fully self-contained module, `Online_Preprocessing_and_Training.py`, that performs the online preprocessing and training over a small subset of the full 2 billion dataset to demonstrate the workload. You can parameterize the same module code to process the full dataset. The \"Scale to 2 billion images\" section below summarizes the changes required to process the full dataset.\n",
+    "This notebook executes a fully self-contained module, `end_to_end.py`, that performs the online preprocessing and training over a small subset of the full 2 billion dataset to demonstrate the workload. You can parameterize the same module code to process the full dataset. The \"Scale to 2 billion images\" section below summarizes the changes required to process the full dataset.\n",
```
Comment (woshiyyya) [inline templates/stable-diffusion-pretraining/03_end_to_end.ipynb:24]:
```suggestion
    "This notebook executes a fully self-contained module, `end_to_end.py`, that performs the online preprocessing and training over a small subset of the full 2 billion dataset to demonstrate the workload. You can parameterize the same module code to process the full dataset. The \"Running production-scale model training\" section below summarizes the changes required to process the full dataset.\n",
```

File: templates/stable-diffusion-pretraining/03_end_to_end.ipynb [hunk 5]
```diff
@@ -61,22 +66,9 @@
     "| 10 | Output path | Change to a permanent path |\n",
     "| 11 | Run the process script | Run as an Anyscale Job |\n",
     "| 12 | First phase of training | Run for 100 epochs |\n",
```
Comment (woshiyyya) [inline templates/stable-diffusion-pretraining/03_end_to_end.ipynb:68]: Oh Actually it's not 100 epochs but 550,000 steps for resolution 256, and 850,000 steps for resolution 512. Approximately 1~2 epochs.

File: templates/stable-diffusion-pretraining/03_end_to_end.ipynb [hunk 6]
```diff
@@ -61,22 +66,9 @@
...
-    }
-   },
-   "source": [
-    "## Want to pre-train with custom data? 📈\n",
-    "\n",
-    "If you're looking to scale your Stable Diffusion pre-training with custom data, we're here to help 🙌 !\n",
+    "| 13 | Second phase of training | Run for 100 epochs loading the checkpoint from the first phase |\n",
     "\n",
-    "👉 **[Check out this link](https://forms.gle/9aDkqAqobBctxxMa8) so we can assist you**."
+    "In terms of infrastructure, you would provision 48 instances of g5.2xlarge instances and 4 instances of p4de.24xlarge instances for the entire process or use Anyscale's autoscaling capabilities to scale up and down as needed."
```
Comment (woshiyyya) [inline templates/stable-diffusion-pretraining/03_end_to_end.ipynb:71]:
```suggestion
    "In terms of infrastructure, you would provision 48 instances of `g5.2xlarge` and 4 instances of `p4de.24xlarge` for the entire process or use Anyscale's autoscaling capabilities to scale up and down as needed."
```

File: templates/stable-diffusion-pretraining/README.ipynb [hunk 7]
```diff
@@ -0,0 +1,107 @@
+{
```
Comment (woshiyyya) [inline templates/stable-diffusion-pretraining/README.ipynb:1]: Will this `README.ipynb` show up at the first place? or `RAEDME.md`?

Comment (marwan116) [reply templates/stable-diffusion-pretraining/README.ipynb:1]: README.ipynb

Comment (marwan116) [reply templates/stable-diffusion-pretraining/README.ipynb:1]: resolving - feel free to re-open if you have any concerns. But I will regenerate README.md once all changes are finalized to keep them in-sync

File: templates/stable-diffusion-pretraining/03_end_to_end.ipynb (hunk 5 above)
Comment (marwan116) [reply templates/stable-diffusion-pretraining/03_end_to_end.ipynb:68]: ok I will update accordingly

**Comment by marwan116** [comment]
@woshiyyya all your comments have been addressed - this should be good to merge from what I can tell, please let me know if there is anything else you'd like me to address

**Review by woshiyyya — APPROVED** [review]

**Pull request merged**